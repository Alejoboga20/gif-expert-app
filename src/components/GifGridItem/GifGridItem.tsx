import { Card } from '@nextui-org/react';

const GifGridItem = ({ title, url }: GifGridItemProps) => {
	return (
		<Card bordered hoverable shadow>
			<Card.Body>
				<Card.Image
					src={url}
					alt={title}
					objectFit='cover'
					width='100%'
					height='10rem'
				></Card.Image>
			</Card.Body>
			<Card.Footer>
				<p>{title}</p>
			</Card.Footer>
		</Card>
	);
};

interface GifGridItemProps {
	title: string;
	url: string;
}

export default GifGridItem;
