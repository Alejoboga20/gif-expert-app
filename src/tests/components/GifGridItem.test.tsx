import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem/GifGridItem';

const title = 'testing';
const url = 'testing';

const benchImg = {
	alt: 'testing',
	className: 'nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image',
	'data-state': 'loading',
	height: '10rem',
	src: 'testing',
	width: '100%',
};

describe('GifGridItem Tests', () => {
	let wrappper = shallow(<GifGridItem title={title} url={url} />);

	beforeEach(() => {
		wrappper = shallow(<GifGridItem title={title} url={url} />);
	});

	test('should Display GifGridItem properly', () => {
		expect(wrappper).toMatchSnapshot();
	});

	test('should have a paragraph with title', () => {
		const paragraph = wrappper.find('p').text();
		expect(paragraph).toBe(title);
	});

	test('should have image set by props', () => {
		const wrappper = mount(<GifGridItem title={title} url={url} />);
		const image = wrappper.find('img');
		expect(image.props()).toMatchObject(benchImg);
	});

	test('should have animate__fadeIn Class', () => {
		const wrappper = mount(<GifGridItem title={title} url={url} />);
		expect(wrappper).toMatchSnapshot();
		const div = wrappper.find('div').at(0);
		const className = div.prop('className');
		expect(className.includes('nextui')).toBe(true);
	});
});
