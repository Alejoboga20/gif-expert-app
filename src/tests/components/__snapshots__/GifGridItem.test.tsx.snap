// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Tests should Display GifGridItem properly 1`] = `
<NextUI - Card
  animated={true}
  bordered={true}
  cover={false}
  hoverable={true}
  preventDefault={false}
  ripple={true}
  shadow={true}
>
  <Styled.div>
    <Memo(Image)
      alt="testing"
      height="10rem"
      objectFit="cover"
      src="testing"
      width="100%"
    />
  </Styled.div>
  <Styled.div>
    <p>
      testing
    </p>
  </Styled.div>
</NextUI - Card>
`;

exports[`GifGridItem Tests should have animate__fadeIn Class 1`] = `
<div
  className="nextui-c-kltynt nextui-c-kltynt-kQghEc-color-default nextui-c-kltynt-iVCgFB-shadow-true nextui-c-kltynt-kUADSW-bordered-true nextui-c-kltynt-bzEnZV-borderWeight-normal nextui-c-kltynt-iNKybL-animated-true nextui-c-kltynt-jQJOMy-hoverable-true nextui-c-kltynt-ifTMkJ-cover-false nextui-c-kltynt-kQghEc-cv"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyDownCapture={[Function]}
  onKeyPress={[Function]}
  onKeyPressCapture={[Function]}
  onKeyUp={[Function]}
  onKeyUpCapture={[Function]}
  role="section"
  tabIndex={-1}
>
  <div
    className="nextui-c-gJBdwd"
  >
    <div
      className="nextui-c-bVLlAJ nextui-c-bVLlAJ-cCitdK-ready-false nextui-c-bVLlAJ-iiVyXnL-css nextui-image-container nextui-image--loading"
      data-state="loading"
    >
      <div
        className="nextui-c-eaDpBR nextui-c-eaDpBR-ibDGmTT-css nextui-image-skeleton"
      />
      <img
        alt="testing"
        className="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
        data-state="loading"
        height="10rem"
        onLoad={[Function]}
        src="testing"
        width="100%"
      />
    </div>
  </div>
  <div
    className="nextui-c-jYFaLZ"
  >
    <p>
      testing
    </p>
  </div>
</div>
`;
